### PR TITLE
Companies

### PR DESCRIPTION
**Pull request by Scandie**

### CODE REVIEW

File: events/companies/forms.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from django import forms
+
+
+class CompanyForm(forms.Form):
```
Comment (ghost) [inline events/companies/forms.py]: Why you don't use ModelForm?

File: events/companies/models.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    @staticmethod
+    def get_by_id(company_id):
+        try:
+            return Company.objects.get(pk=company_id)
+        except Company.DoesNotExist:
+            return None
+
+    @staticmethod
+    def get_teams(company_id):
+        return [team.name for team in Company.get_by_id(company_id).teams.all()]
```
Comment (VKotsovskyi) [inline events/companies/models.py]: Why do you need only names?

File: events/companies/models.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+    @staticmethod
+    def get_user_company(request):
+        company = Company.objects.filter(admin=request.user).first()
```
Comment (VKotsovskyi) [inline events/companies/models.py:30]: Why didn't you use `get` inside of `try-except`? Just curious your opinion:))

File: events/companies/views.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+                return PERMISSION_DENIED
+            companies = Company.get_all()
+            response = {"companies": [model_to_dict(company) for company in companies]}
+            return JsonResponse(response, status=200)
+
+        company = Company.get_by_id(company_id)
+        if not company:
+            return COMPANY_NOT_EXISTS
+        if Company.get_user_company(request) != company:
+            if not request.user.is_superuser:
```
Comment (VKotsovskyi) [inline events/companies/views.py]: You can unite this two `if`s into one with `and` as far as I can see.

File: events/companies/views.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+    def delete(self, request, company_id):
+        company = Company.get_by_id(company_id)
+        if not CompanyView.check_company_rights(request, company):
+                return PERMISSION_DENIED
+        company.delete()
+        return NO_CONTENT
+
+    @staticmethod
+    def check_company_rights(request, company):
```
Comment (VKotsovskyi) [inline events/companies/views.py:81]: I would say this is private method. Are you going to use it in any other Classes?

File: events/companies/views.py [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+        if not request.user.is_superuser:
+            return company.admin.id == request.user.id
+        return True
+
+    @staticmethod
+    def validate_admin_field(user, errors, data, company=None):
+        if not data.get('admin'):
+            errors['admin'] = ['This field is required']
+        else:
```
Comment (VKotsovskyi) [inline events/companies/views.py]: Why didn't you used `elif`?

File: events/companies/views.py [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+        return errors
+
+
+class TeamView(View):
+
+    def get(self, request, company_id, team_id=None):
+        company = Company.get_by_id(company_id)
+        if not company:
+            return COMPANY_NOT_EXISTS
+        if Company.get_user_company(request) != company:
```
Comment (VKotsovskyi) [inline events/companies/views.py]: Looks like there are few more places where you can use `and` instead of two `if`s

**Review by VKotsovskyi — CHANGES_REQUESTED** [review]

File: events/companies/models.py (hunk 3 above)
Comment (Scandie) [reply events/companies/models.py:30]: @VKotsovskyi Can't really say. Would the `try-except` match better for this purpose?

File: events/companies/views.py (hunk 5 above)
Comment (Scandie) [reply events/companies/views.py:81]: @VKotsovskyi By now `check_company_rights` method is used in all of the three views.

Comment (VKotsovskyi) [reply events/companies/views.py:81]: +

**Review by VKotsovskyi — APPROVED** [review]
Good Job 👍

**Review by ghost — APPROVED** [review]

**Pull request merged**